### PR TITLE
Convert DateResponse to Kotlin

### PR DESCRIPTION
**Pull request by JSunde**

### CODE REVIEW

**Comment by JSunde** [comment]
PTAL @gino-m @scolsen?

**Review by gino-m — APPROVED** [review]

File: ground/src/main/java/com/google/android/ground/model/submission/DateResponse.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import kotlinx.serialization.Serializable
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ * A user-provided response to a date question task.
+ */
+@Serializable
+class DateResponse(val date: @Contextual Date) : Response {
```
Comment (scolsen) [inline ground/src/main/java/com/google/android/ground/model/submission/DateResponse.kt:29]: Should we make this a data class?

**Review by scolsen — APPROVED** [review]

File: ground/src/main/java/com/google/android/ground/model/submission/DateResponse.kt (hunk 1 above)
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/model/submission/DateResponse.kt]: Good idea, I'm not sure I realized that data classes handled the equals() toString() and hashCode() methods so this ended up much cleaner after that!

Comment (scolsen) [reply ground/src/main/java/com/google/android/ground/model/submission/DateResponse.kt]: awesome!

**Review by scolsen — APPROVED** [review]

**Pull request merged**